_posts/2020-02-11-spring-boot-openapi.md

```diff
@@ -0,0 +1,394 @@
+```
+
+* First the [`ref`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields-7) field, allows us to refer own defined Schema. In this case the User schema.
+* Then the `summary`, which is optional and is intended to apply to all operations in this path.
+* Next is the [`operationId`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields-8), which is a unique string and is used to identify the operation. Think about it as our method name.
+* Last the [`responses`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#responsesObject) object.It must define at least one successful response code for an operation call. 
+
+### Components
+
+The objects of the API are all described here. The objects defined within the components object will have no effect on the API unless they are explicitly referenced from properties outside the components object.
+
+```yaml
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        username:
+          type: string
+        firstName:
+          type: string
+        lastName:
+          type: string
+        email:
+          type: string
+        password:
+          type: string
+        phone:
+          type: string
+        userStatus:
+          type: integer
+          description: User Status
+          format: int32
+  securitySchemes:
+    reflectoring_auth:
+      type: oauth2
+      flows:
+        implicit:
+          authorizationUrl: http://reflectoring.swagger.io/oauth/dialog
+          scopes:
+            write:users: modify users
+            read:users: read users
+    api_key:
+      type: apiKey
+      name: api_key
+      in: header
+```
+
+* First the [`schemas`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#schema-object) field, allows the definition of our DTOs.
+* And last the [`securitySchemes`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#security-scheme-object), and it defines a security scheme that can be used by the operations.
+
+## Generating Code From an API Specification
+
+Having defined an API, we'll now create code from the YAML document above. 
+
+We'll take a look at two different approaches to generating the code:
+
+1. Using [Swagger Editor](http://editor.swagger.io/) to generate code manually, and
+2. Using the [OpenAPI Maven plugin](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin) to generate code from a Maven build.
+
+### Generating Code from Swagger Editor
+
+Although that's an approach which I wouldn't take, I must introduce it. 
+
+So, let's go over to [Swagger Editor](http://editor.swagger.io/) and paste our `<specification>.yml` into it. Then, we select *Generate Server* from the menu and pick what kind of a server we'd would like to generate (I went with Spring).
```

Comment:
Can you link to the YAML file so that readers can try this out themselves with your YAML file?